rt/rs/microprofile-client/src/main/java/org/apache/cxf/microprofile/client/MicroProfileClientFactoryBean.java

```diff
@@ -63,7 +66,7 @@ public MicroProfileClientFactoryBean(MicroProfileClientConfigurableImpl<RestClie
         super(new MicroProfileServiceFactoryBean());
         this.configuration = configuration.getConfiguration();
         this.comparator = MicroProfileClientProviderFactory.createComparator(this);
-        this.executorService = executorService;
+        this.executorService = (executorService == null) ? defaultExecutorService() : executorService; 
```

Comment:
maybe make it lazy to not create it (forkjoin common pool) if the app does not use it and the client for its lifecycle?

Reply:
Thanks for the review! This one is very tricky to make lazy. Essentially, this `executorService` sets (or unsets) the CXF's `EXECUTOR_SERVICE_PROPERTY` which basically controls the how the async client invocation should be handled (it is used by MP and other clients). If we don't provide `executorService` upfront, the property won't be set and async callback (`MPRestClientCallback`) is going to be called in the context of CXF's work queue pool. The common `ForkJoinPool` is present 99.9% of the time.

Reply:
es = new LazyExecutor();

lazyexecutor 100% delegates to common one, it just enables to not create threads until it is done so the pool is async, not the facing instance ;)

Reply:
Sorry, this part I didn't get. With respect to threads (aka workers), `ForkJoinPool` is 100% lazy: it does not preallocate threads upfront if there is no work to be done, the `ForkJoinPool`'s threads are created only when work is scheduled.

Reply:
Yep, was not a good phrasing choice. Fact is getting triggers the common (var) instantiation which brings a lot of work - for nothing in some cases and that's what I'd like to avoid.
Having a facade enables to just load a class and do a new without dependencies (static init) and speed up the startup.
We faced it in openwebbeans and this is why we have that wrapper https://github.com/apache/openwebbeans/blob/9b04c5e906294bcad7c61f30b5c5fcc8956b2f17/webbeans-impl/src/main/java/org/apache/webbeans/event/NotificationManager.java#L954.

Reply:
:+1: thanks for bearing with me, the concern is clear now, fixed!